src/ring_theory/discrete_valuation_ring.lean

```diff
@@ -367,6 +379,96 @@ begin
   { apply (hirr.ne_zero (pow_eq_zero h)).elim, }
 end
 
+/-!
+## The additive valuation on a DVR
+-/
+
+/-- The `ℕ`-valued additive valuation on a DVR (returns junk at `0` rather than `+∞`) -/
+noncomputable def add_val (R : Type u) [integral_domain R] [discrete_valuation_ring R] : R → ℕ :=
+λ r, if hr : r = 0 then 0 else
+  classical.some (associated_pow_irreducible hr (classical.some_spec $ exists_irreducible R))
+
+theorem add_val_spec {r : R} (hr : r ≠ 0) :
```

Comment:
`add_val_def` is the more friendly version intended for general use, right?
```suggestion
/-- `add_val_spec` is a specific instantiation of `add_val_def` used to prove the latter. -/
theorem add_val_spec {r : R} (hr : r ≠ 0) :
```

Reply:
Yes, `add_val_def` is the useful version. Even though it's not the actual definition, it's the "mathematician's definition" of the additive valuation. I'm not sure if `add_val_spec` is a specific instantiation of `add_val_def` -- I'm not really sure I understand what that means.